### ISSUE TITLE
pdfmake JSON schema ?

### ISSUE THREAD

**Issue by dehnhard** [bpampuch/pdfmake#1408]
In addition to a new documentation (#1238) a JSON schema file would be helpful for developers using pdfmake. In a modern IDE like WebStorm it can be used for validation and code completion in the editor.

**Comment by knakamatsu** [comment]
Hello, @liborm85 .
Always thank you for a wonderful library.

I attempted to create a JSON Schema, but the available properties are written in various places in the source code, and it is difficult to grasp everything.

So how about creating a new file that defines all the properties name?It is like a constant file.
This is useful to make a good document, i think.
If you think good too, I'll try.

**Comment by liborm85** [comment]
Yes, it is now complicated to find what properties are used and where. Is need  create new documentation, I've already created [issue](https://github.com/bpampuch/pdfmake/issues/1238).

Simple "Cheat Sheet" is available from [this post in issue](https://github.com/bpampuch/pdfmake/issues/1238#issuecomment-423785361).

**Comment by knakamatsu** [comment]
Thanks a lot.
This issue labeled "Feature request.", so I’ll give it a try JSON Schema based on Cheat Sheet.
